src/map/chrif.c

```diff
@@ -324,6 +324,9 @@ bool chrif_save(struct map_session_data *sd, int flag) {
 	if( sd->save_quest )
 		intif->quest_save(sd);
 
+	if (VECTOR_LENGTH(sd->storage))
```

Comment:
You likely want to save the storage even if it's empty (to handle the case where the storage was full, then items get deleted from it)